commit 571ae2424715009320db7a61e3d6274589cb18a3
Author: Frankie Bagnardi <f.bagnardi@gmail.com>
Date:   Tue Dec 9 22:33:12 2014 -0700

    memoize parseTagSpec

diff --git a/index.js b/index.js
index 4bcc946..34d0ca1 100644
--- a/index.js
+++ b/index.js
@@ -15,6 +15,8 @@ ParseError.prototype = Object.create(Error.prototype)
 ParseError.prototype.name = 'JSnoXParseError'
 
 
+var specCache = {};
+
 // Convert a tag specification string into an object
 // eg. 'input:checkbox#foo.bar[name=asdf]' produces the output:
 // {
@@ -29,6 +31,8 @@ ParseError.prototype.name = 'JSnoXParseError'
 function parseTagSpec(specString) {
     if (!specString.match) throw new ParseError(specString) // We didnt' receive a string
 
+    if (specCache[specString]) return specCache[specString];
+
     // Parse tagName, and optional type attribute
     var tagMatch = specString.match(tagNameRegex)
     if (!tagMatch) throw new ParseError(specString)
@@ -52,10 +56,15 @@ function parseTagSpec(specString) {
         }
     })
     if (classes.length) props.className = classes.join(' ')
-    return {
+
+    var spec = {
         tagName: tagName,
         props: props
-    }
+    };
+
+    specCache[specString] = spec
+
+    return spec
 }
 
 // Merge two objects, producing a new object with their properties
